contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,959 @@
+         * If estimateAmount1 call fails it could be due to _currentPrice being really
+         * close to a tick and amount1 too big to compute.
+         *
+         * If token addresses were reversed estimateAmount0 would be required here
+         */
+        uint256 _normalizedWethAmount = 1 ether;
+        uint256 _correspondingOethAmount = helper.estimateAmount1(
+            _normalizedWethAmount,
+            address(0), // no need to pass pool address when current price is specified
+            _currentPrice,
+            lowerTick,
+            upperTick
+        );
+
+        // 18 decimal number expressed weth tick share
+        uint256 _wethSharePct = _normalizedWethAmount.divPrecisely(
+            _normalizedWethAmount + _correspondingOethAmount
+        );
+
+        if (
+            _wethSharePct >= allowedWethShareStart &&
+            _wethSharePct <= allowedWethShareEnd
+        ) {
+            emit PoolRebalanced(_wethSharePct);
+        } else {
+            revert PoolRebalanceOutOfBounds(
+                _wethSharePct,
+                allowedWethShareStart,
+                allowedWethShareEnd
+            );
+        }
+    }
+
+    /**
+     * Burns any OETHb tokens remaining on the strategy contract
+     */
+    function _burnOethbOnTheContract() internal {
+        uint256 _oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        if (_oethbBalance > 0) {
+            IVault(vaultAddress).burnForStrategy(_oethbBalance);
+        }
+    }
+
+    /// @dev this function assumes there are no uncollected tokens in the clPool owned by the.
+    ///      strategy contract. For that reason any liquidity withdrawals also collect the tokens.
+    function _updateUnderlyingAssets() internal {
+        if (tokenId == 0) {
+            underlyingAssets = 0;
+            emit UnderlyingAssetsUpdated(underlyingAssets);
+            return;
+        }
+
+        uint128 _liquidity = _getLiquidity();
+
+        /**
+         * Our net value represent the smallest amount of tokens we are able to extract from the position
+         * given our liquidity.
+         *
+         * The least amount of tokens extraditable from the position is where the active trading price is
+         * at the ticker 0 meaning the pool is offering 1:1 trades between WETH & OETHb. At that moment the pool
+         * consists completely of OETHb and no WETH.
+         *
+         * The more swaps from WETH -> OETHb happen on the pool the more the price starts to move towards the -1
+         * ticker making OETHb (priced in WETH) more expensive.
+         */
+        (uint256 _wethAmount, uint256 _oethbAmount) = helper
+            .getAmountsForLiquidity(
+                sqrtRatioX96TickClosestToParity, // sqrtRatioX96
+                sqrtRatioX96TickLower, // sqrtRatioAX96
+                sqrtRatioX96TickHigher, // sqrtRatioBX96
+                _liquidity
+            );
+
+        require(_wethAmount == 0, "Non zero wethAmount");
```

Comment:
🔴 Is this going to fail after a withdrawAll? There will still be a token id, but liquidity and thus `_wethAmount` will both be zero?

Reply:
On 0 liquidity passed to the Sugar helper the function just returns 0 for both `wethAmount` & `_oethbAmount`. Which I think is the correct behaviour. And consequently `underlyingAssets` is correctly set to 0. 

IMO this doesn't require any code changes.

Reply:
I totally forgot what this was doing. My bad. Carry on.